packages/core/contracts/oracle/implementation/SkinnyOptimisticOracle.sol

```diff
@@ -0,0 +1,716 @@
+    ) private pure returns (bytes32) {
+        return keccak256(abi.encode(_requester, _identifier, _timestamp, _ancillaryData));
+    }
+
+    // Returns hash of request parameters. These are mapped to the unique request ID to track a request's lifecycle.
+    function _getRequestHash(Request memory _request) private pure returns (bytes32) {
+        return keccak256(abi.encode(_request));
+    }
+
+    // Resolves a price request that has expired or been disputed and a price is available from the DVM. This will
+    // revert if the unique request ID does not match the hashed request parameters. This also marks the request
+    // as settled, therefore this method can only be triggered once per eligible request.
+    function _settle(
+        address _requester,
+        bytes32 _identifier,
+        uint256 _timestamp,
+        bytes memory _ancillaryData,
+        Request memory _request
+    ) private returns (uint256 payout, int256 resolvedPrice) {
+        bytes32 requestId = _getId(_requester, _identifier, _timestamp, _ancillaryData);
+        _validateRequestHash(requestId, _request);
+
+        // Associate settled request params with ID.
+        Request memory settledRequest =
+            Request({
+                proposer: _request.proposer,
+                disputer: _request.disputer,
+                currency: _request.currency,
+                settled: true, // Modified
+                proposedPrice: _request.proposedPrice,
+                resolvedPrice: _request.resolvedPrice,
+                expirationTime: _request.expirationTime,
+                reward: _request.reward,
+                finalFee: _request.finalFee,
+                bond: _request.bond,
+                customLiveness: _request.customLiveness
+            });
+
+        OptimisticOracleInterface.State state =
+            _getState(_requester, _identifier, _timestamp, _ancillaryData, _request);
+        if (state == OptimisticOracleInterface.State.Expired) {
+            // In the expiry case, just pay back the proposer's bond and final fee along with the reward.
+            resolvedPrice = settledRequest.proposedPrice;
+            settledRequest.resolvedPrice = resolvedPrice;
+            payout = settledRequest.bond.add(settledRequest.finalFee).add(settledRequest.reward);
+            settledRequest.currency.safeTransfer(settledRequest.proposer, payout);
+        } else if (state == OptimisticOracleInterface.State.Resolved) {
+            // In the Resolved case, pay either the disputer or the proposer the entire payout (+ bond and reward).
+            resolvedPrice = _getOracle().getPrice(
+                _identifier,
+                _timestamp,
+                _stampAncillaryData(_ancillaryData, _requester)
+            );
+            settledRequest.resolvedPrice = resolvedPrice;
+            bool disputeSuccess = settledRequest.resolvedPrice != settledRequest.proposedPrice;
+            uint256 bond = settledRequest.bond;
```

Comment:
nit: you can remove this variable and just reference `settledRequest.bond` on lines 607 and 614 thereby saving the declaration of a variable.